src/App.tsx

```diff
@@ -1,223 +1,298 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useContext,
+  useEffect,
+  useState,
+} from 'react';
+import classNames from 'classnames';
+import { TodoList } from './components/TodoList';
+import {
+  getTodos,
+  deleteTodo,
+  patchTodo,
+  patchTitleTodo,
+} from './api/todos';
+import { Todo } from './types/Todo';
+import { ErrorNotification } from './components/ErrorNotification';
+import { SortType } from './types/filterBy';
+import { NewTodoField } from './components/NewTodoField';
 import { AuthContext } from './components/Auth/AuthContext';
+import { ActiveTodos } from './components/ActiveTodos';
+
+function filterTodos(
+  todos: Todo[],
+  sortType: SortType,
+) {
+  const visibleTodos = [...todos];
+
+  switch (sortType) {
+    case SortType.Active:
+      return visibleTodos.filter(todo => !todo.completed);
+
+    case SortType.Completed:
+      return visibleTodos.filter(todo => todo.completed);
+    default:
+      return visibleTodos;
+  }
+}
 
 export const App: React.FC = () => {
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
-  const newTodoField = useRef<HTMLInputElement>(null);
+
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [sortType, setSortType] = useState<SortType>(SortType.All);
+  const [selectedLink, setSelectedLink] = useState<string>('All');
+  const [errorMessage, setErrorMessage] = useState<string>('');
+  const [activeItems, setActiveItems] = useState<number>(0);
+  const [visibleLoader, setVisibleLoader] = useState(false);
+
+  let userId = 0;
+
+  if (user?.id) {
+    userId = user.id;
+  }
 
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
-    if (newTodoField.current) {
-      newTodoField.current.focus();
-    }
+    getTodos(userId)
+      .then(todosFromServer => {
+        setTodos(todosFromServer);
+        setActiveItems((prevItems) => {
+          const tempTodos = [...todosFromServer];
+          let tempItems = prevItems;
+
+          tempTodos.forEach(todo => {
+            if (!todo.completed) {
+              tempItems += 1;
+            }
+          });
+
+          setTodos(tempTodos);
+
+          return tempItems;
+        });
+      })
+      .catch(() => setErrorMessage('Unable to update todos'));
```

Comment:
better to create an enum with errors

Reply:
I think that my errorNotification also good